src/KeyCommand.web.js

```diff
@@ -0,0 +1,152 @@
+ * @param {Function} callback - Callback to be called when the event is triggered.
+ * @returns {Object}
+ */
+EventEmitter.addListener = (type, callback) => {
+    EventEmitter.on(type, callback);
+    return {
+        remove: () => EventEmitter.removeListener(type, callback),
+    };
+};
+
+const commands = [];
+const constants = {
+    keyInputDownArrow: 'keyInputDownArrow',
+    keyInputEscape: 'keyInputEscape',
+    keyInputLeftArrow: 'keyInputLeftArrow',
+    keyInputRightArrow: 'keyInputRightArrow',
+    keyInputUpArrow: 'keyInputUpArrow',
+    keyModifierCapsLock: 'keyModifierCapsLock',
+    keyModifierCommand: 'keyModifierCommand',
+    keyModifierControl: 'keyModifierControl',
+    keyModifierControlCommand: 'keyModifierControlCommand',
+    keyModifierControlOption: 'keyModifierControlOption',
+    keyModifierControlOptionCommand: 'keyModifierControlOptionCommand',
+    keyModifierNumericPad: 'keyModifierNumericPad',
+    keyModifierOption: 'keyModifierOption',
+    keyModifierOptionCommand: 'keyModifierOptionCommand',
+    keyModifierShift: 'keyModifierShift',
+    keyModifierShiftCommand: 'keyModifierShiftCommand',
+};
+
+/**
+ * Gets modifiers from a keyboard event.
+ *
+ * @param {Event} event
+ * @returns {Array<String>}
+ */
+function getKeyEventModifiers(event) {
+    const modifiers = [];
+    if (event.shiftKey) {
+        return constants.keyModifierShift;
+    }
+    if (event.ctrlKey) {
+        return constants.keyModifierControl;
+    }
+    if (event.altKey) {
+        return constants.keyModifierOption;
+    }
+    if (event.metaKey) {
+        return constants.keyModifierCommand;
+    }
+    return modifiers;
+}
+
+/**
+ * Finds index of register key command and listen to the event.
+ *
+ * @param {Object} json - List of key command objects.
+ * @param {string} json.input - any character key from the keyboard.
+ * @param {number} json.modifierFlags - predefined command from getConstants enum.
+ *
+ * @returns {number} index of registered keyCommand.
+ */
+function registeredCommandIndex(json) {
+    const matchesModifierFlags = item => (
+        item.modifierFlags === json.modifierFlags
+        || (_.isEmpty(item.modifierFlags) && _.isEmpty(json.modifierFlags))
+    );
+
+    const matchesEscape = item => (item.input === constants.keyInputEscape && json.input === 'escape');
+
+    return _.findIndex(commands, item => (
+        item.input === json.input
+        && matchesModifierFlags(item)
+    ) || matchesEscape(item));
+}
+
+/**
+ * Checks if event exists in registered key command array and returns index.
+ *
+ * @param {Event} event
+ * @returns {number} index of registered keyCommand.
+ */
+function eventMatchesInput(event) {
+    const modifierFlags = getKeyEventModifiers(event);
+    return registeredCommandIndex({input: event.key.toLowerCase(), modifierFlags});
+}
+
+function onKeyDown(event) {
+    const index = eventMatchesInput(event);
+    if (index === -1) {
+        return;
+    }
```

Comment:
```suggestion
    if (!eventMatchesInput(event)) {
        return;
    }
```

Reply:
-1 stands for "not found". 0 stands for position.

Reply:
Yup, that's why I suggested a change to the function.

Reply:
ah, missed other comment.

Reply:
wait, we need an `index` below. this suggestion doesn't work.